src/rdkafka_conf.c

```diff
@@ -4304,25 +4363,107 @@ int unittest_conf (void) {
                                 "!1.2.3.4.5!!! a", NULL, 0);
         RD_UT_ASSERT(res == RD_KAFKA_CONF_OK, "%d", res);
 
+        /* Verify that sasl.oauthbearer.* string-safing works */
+        res = rd_kafka_conf_set(conf, "sasl.oauthbearer.method",
+                                "oidc", NULL, 0);
+        RD_UT_ASSERT(res == RD_KAFKA_CONF_OK, "%d", res);
+
+        res = rd_kafka_conf_set(conf, "sasl.oauthbearer.client.id",
+                                "randomuniqclientid", NULL, 0);
+        RD_UT_ASSERT(res == RD_KAFKA_CONF_OK, "%d", res);
+
+        res = rd_kafka_conf_set(conf, "sasl.oauthbearer.client.secret",
+                                "randomuniqclientsecret", NULL, 0);
+        RD_UT_ASSERT(res == RD_KAFKA_CONF_OK, "%d", res);
+
+        res = rd_kafka_conf_set(conf, "sasl.oauthbearer.scope",
+                                "svc-devel-C348S-pawneeTech", NULL, 0);
+        RD_UT_ASSERT(res == RD_KAFKA_CONF_OK, "%d", res);
+
+        res = rd_kafka_conf_set(conf, "sasl.oauthbearer.extensions",
+                                "supportFeatureX=true", NULL, 0);
+        RD_UT_ASSERT(res == RD_KAFKA_CONF_OK, "%d", res);
+
+        res = rd_kafka_conf_set(conf, "sasl.oauthbearer.token.endpoint.url",
+                                "https://dev-531534.okta.com/oauth2/token",
```

Comment:
Use something that does not risk creating outbound connections. E.g. `https://localhost:1/token`